src/mode/modeHandler.ts

```diff
@@ -2063,13 +2063,16 @@ export class ModeHandler implements vscode.Disposable {
   }
 
   setStatusBarColor(color: string): void {
+    const currentColorCustomizations = vscode.workspace
+      .getConfiguration('workbench')
+      .get('colorCustomizations');
     vscode.workspace.getConfiguration('workbench').update(
       'colorCustomizations',
-      {
+      Object.assign({}, currentColorCustomizations, {
```

Comment:
I've never used `Object.assign` before, is there a difference between:

1) `Object.assign({}, currentColor....)` and
2) `Object.assign(currentColor..., `

ie. why do you need the initial `{}`?

Reply:
The first parameter of the `Object.assign` function is the target object. Like you wrote below, it works like an upsert, where from left to right properties get merged.

This is the behavior that I intend, since I want to get a copy of the color customizations and then overwrite its properties regarding the status bar.

The target object gets modified by the `Object.assign` call, that's why I pass an empty object at first. But indeed, in this case, this is not needed. I will remove it.